test/Chainweb/Test/SPV.hs

```diff
@@ -101,6 +118,140 @@ targetChain c srcBlock = do
+        S.each (zip [0..] $ fmap fst $ toList $ _payloadWithOutputsTransactions pay)
+            & S.map (\(b,c) -> (h,n,b,c))
+
+    -- Given
+    -- - block header,
+    -- - tx count of block,
+    -- - tx index,
+    -- - tx, and
+    -- - target chain,
+    --
+    -- creates and verifies SPV proofs and returns list of
+    -- - tx proof length
+    -- - block size (tx count)
+    -- - block height difference
+    -- - distance between source chain and target chain
+    -- - size of tx
+    --
+    go
+        :: PayloadCas cas
+        => CutDb cas
+        -> (BlockHeader, Int, Int, Transaction, ChainId)
+        -> IO (Maybe [Double])
+    go cutDb (h, n, txIx, tx, trgChain) = do
+
+        let inner = do
+                -- create inclusion proof for transaction
+                proof <- createTransactionProof cutDb trgChain
+                    (_chainId h) -- source chain
+                    (_blockHeight h) -- source block height
+                    txIx -- transaction index
+                subj <- verifyTransactionProof cutDb proof
+                assertEqual "transaction proof subject matches transaction" tx subj
+
+                -- return (proof size, block size, height, distance, tx size)
+                return
+                    [ int $ BL.length $ encode proof
+                    , int n
+                    , int $ _blockHeight h
+                    , int $ distance cutDb h trgChain
+                    , int $ B.length (_transactionBytes tx)
+                    ]
+
+        isReachable <- reachable cutDb h trgChain
+        try inner >>= \case
+            Right x -> do
+                assertBool "SPV proof creation succeeded although target chain is not reachable" isReachable
+                return (Just x)
+            Left SpvExceptionTargetNotReachable{} -> do
+                assertBool "SPV proof creation failed although target chain is reachable" (not isReachable)
+                return Nothing
+            Left e -> throwM e
+
+    -- Distance between source chain an target chain
+    --
+    distance cutDb h trgChain = length
+        $ shortestPath (_chainId h) trgChain
+        $ _chainGraph cutDb
+
+    -- Check whether target chain is reachable from the source block
+    --
+    reachable :: CutDb as -> BlockHeader -> ChainId -> IO Bool
+    reachable cutDb h trgChain = do
+        m <- maxRank $ cutDb ^?! cutDbBlockHeaderDb trgChain
+        return $ (int m - int (_blockHeight h)) >= (distance cutDb h trgChain)
+
+    -- regression model with @createTransactionProof@. Proof size doesn't depend on target height.
+    --
+    regress r
+        | [proofSize, blockSize, _heightDiff, chainDist, txSize] <- V.fromList <$> L.transpose r
+            = olsRegress [V.map (logBase 2) blockSize, chainDist, txSize] proofSize
+        | otherwise = error "Chainweb.Test.SPV.spvTest.regress: fail to match regressor list. This is a bug in the test code."
```

Comment:
if there is any chance a maintainer can screw this up, it should be an `assertionFailed` (I can't remember the exact name) with output that gives a clue what went wrong, same for `regressWithHeightDiff`

Reply:
If this fails, it's about the test itself and not about the tested code. 

It's really just about preventing an incomplete pattern match warning in the code for computing the statistics. The regressor value rows are collected in a list (instead of a tuple) because that's what `L.transpose` expects.